flipperdevices/flipperzero-firmware
# NFC: Auto-run reading scripts

# What's new

- "Read card" has been renamed and will automatically run corresponding reading script.
- Old functionality has been moved to reading scripts menu under "Generic NFC."
- Improve navigation flow for MF Ultralight as proof of concept.
- Replace MF Desfire menu with save button & fix scene states under MF Classic menu.
- Move MF Ultralight menu into success scene.

## Todo

- Move remaining submenus into success scenes.
- Apply same back/retry navigation to other card types.
- Test saving with a real MF Desfire card (I don't have any).
- Suppress success notification if returning from save/emulate menus.

## Notes
This PR is meant to be a proof of concept for aforementioned changes. I'd like to know if automatically running reading scripts with the universal read option is better or worse than what happens currently. My motivation is that I've seen various posts of confused users who didn't realize to fully read a card they had to run the corresponding script.  
_I'd like some people to flash this and provide feedback whether or not these changes improve the flow of NFC scenes._

# Verification 

- Build & Flash
- Open NFC app and run the top "Read NFC Tag" option.
- Tap any supported NFC tag to the device.
- Corresponding reading app should automatically run.
- Original functionality is under Reading Scripts > Read Generic NFC.

# Checklist (For Reviewer)
